### PR TITLE
specs, use union for extensible enum

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
1. use `union` for current "extensible enum"

<details>
<summary>Dependency version</summary>
Need to use Azure.Core 0.53.1.

0.53.0 has such validation error

```
error: ✘ Union contains non-string value type Scalar.: C:/github/cadl-ranch/node_modules/.pnpm/@typespec+compiler@0.53.0/node_modules/@typespec/compiler/lib/lib.tsp:90:1
error: ✘ Union contains non-string value type Scalar.: C:/github/cadl-ranch/node_modules/.pnpm/@typespec+compiler@0.53.0/node_modules/@typespec/compiler/lib/lib.tsp:90:1
error: ✘ The operation linked in  @pollingOperation must return a valid status monitor.  The status monitor model must contain a 'status' property, or a property decorated with  '@lroStatus'.  The status field must be of Enum or Union type and contain terminal status values for success and failure.: C:/github/cadl-ranch/packages/cadl-ranch-specs/http/azure/core/lro/rpc-legacy/main.tsp:142:3
error: ✘ ✘ Scenario azure/core/lro/rpc-legacy is invalid.
C:\github\cadl-ranch\packages\cadl-ranch-specs:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @azure-tools/cadl-ranch-specs@0.30.0 validate-scenarios: `cadl-ranch validate-scenarios ./http`
```
</details>

# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/type/enum/extensible/main.tsp [hunk 1]
```diff
@@ -7,27 +7,29 @@ using TypeSpec.Http;
 namespace Type.Enum.Extensible;
 
 @doc("Days of the week")
-enum DaysOfWeekExtensibleEnum {
+union DaysOfWeekExtensibleEnum {
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/type/enum/extensible/main.tsp:10]: `getKnownValue(): DaysOfWeekExtensibleEnum;` likely have problem even in TCGC.

tsp compiler would think this be multiple responses. (`HttpOperationResponse` which contains multiple `HttpOperationResponseContent` in `responses` property)

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/type/enum/extensible/main.tsp:10]: Left a message in https://github.com/microsoft/typespec/issues/2848#issuecomment-1952088905

It is pretty hard for emitter to handle this case, when the Union is already taken apart to Type.

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/type/enum/extensible/main.tsp:10]: Changed to use explicit `@body`
```ts
  getKnownValue(): {
    @body body: DaysOfWeekExtensibleEnum;
  };
```
on Timothee's advice. This should unblock emitter.

File: packages/cadl-ranch-specs/http/type/enum/fixed/main.tsp [hunk 2]
```diff
@@ -8,8 +8,6 @@ using TypeSpec.Http;
 @scenarioService("/type/enum/fixed")
 namespace Type.Enum.Fixed;
 
-#suppress "@azure-tools/typespec-azure-core/use-extensible-enum" "For testing"
-@fixed
```
Comment (lmazuel) [inline packages/cadl-ranch-specs/http/type/enum/fixed/main.tsp:12]: Too soon for this change, see derails in long email I'm writing right now

File: packages/cadl-ranch-specs/http/type/property/value-types/main.tsp [hunk 3]
```diff
@@ -119,17 +119,17 @@ interface Duration extends ModelOperations<DurationProperty, "P123DT22H14M12.011
 
 // Test a model with an enum property
 @doc("Enum that will be used as a property for model EnumProperty. Non-extensible.")
-enum InnerEnum {
+union InnerEnum {
```
Comment (lmazuel) [inline packages/cadl-ranch-specs/http/type/property/value-types/main.tsp:122]: The doc says explicitly this is non-extensible, we should add `@fixed` instead

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/type/property/value-types/main.tsp:122]: There is a `FixedInnerEnum` after this model. So I tend to think the doc was wrong. I would update the doc.

**Review by msyyc — APPROVED** [review]

**Pull request merged**